src/libDirectoryService/ViewChangePostProcessing.cpp

```diff
@@ -229,16 +226,22 @@ void DirectoryService::ProcessViewChangeConsensusWhenDone() {
     return;
   }
 
-  unsigned int nodeToSendToLookUpLo = m_mediator.GetShardSize(true) / 4;
-  unsigned int nodeToSendToLookUpHi =
-      nodeToSendToLookUpLo + TX_SHARING_CLUSTER_SIZE;
-
-  if (m_consensusMyID > nodeToSendToLookUpLo &&
-      m_consensusMyID < nodeToSendToLookUpHi) {
-    m_mediator.m_lookup->SendMessageToLookupNodes(vcblock_message);
-    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
-              "I the part of the subset of DS committee that have sent the "
-              "VCBlock to the lookup nodes");
+  // Broadcasting vcblock to lookup nodes iff view change do not occur before ds
+  // block consensus. This is to be consistent with how normal node process the
+  // vc block (before ds block).
+  if (viewChangeState != DSBLOCK_CONSENSUS &&
+      viewChangeState != DSBLOCK_CONSENSUS_PREP) {
+    unsigned int nodeToSendToLookUpLo = m_mediator.GetShardSize(true) / 4;
```

Comment:
This logic fails in general as we observed earlier if ds committe size is smaller than shard_size/4. (Ex. Ds committee with 100 nodes and Shard-Size of 500). I think this should be fine for this PR, but we need to fix this logic in general wherever applicable.

Reply:
Created an issue https://github.com/Zilliqa/Issues/issues/226 to track this in future.